rpc/v8/subscriptions.go

```diff
@@ -216,35 +216,41 @@ func (h *Handler) SubscribeEvents(ctx context.Context, fromAddr *felt.Felt, keys
 	eventsPreviouslySent := make(map[SentEvent]struct{})
 
 	subscriber := subscriber{
-		onStart: func(ctx context.Context, id uint64, _ any) error {
+		onStart: func(ctx context.Context, id uint64, sub *subscription, _ any) error {
 			return h.processEvents(ctx, w, id, requestedHeader.Number, headHeader.Number, fromAddr, keys, nil)
 		},
-		onReorg: func(ctx context.Context, id uint64, reorg *sync.ReorgBlockRange) error {
+		onReorg: func(ctx context.Context, id uint64, sub *subscription, reorg *sync.ReorgBlockRange) error {
 			if err := sendReorg(w, reorg, id); err != nil {
 				return err
 			}
 			nextBlock = reorg.StartBlockNum
 			return nil
 		},
-		onNewHead: func(ctx context.Context, id uint64, head *core.Block) error {
+		onNewHead: func(ctx context.Context, id uint64, sub *subscription, head *core.Block) error {
 			if err := h.processEvents(ctx, w, id, nextBlock, head.Number, fromAddr, keys, eventsPreviouslySent); err != nil {
 				return err
 			}
 			nextBlock = head.Number + 1
 			return nil
 		},
-		onPending: func(ctx context.Context, id uint64, pending *core.Block) error {
+		onPending: func(ctx context.Context, id uint64, sub *subscription, pending *core.Block) error {
 			return h.processEvents(ctx, w, id, nextBlock, nextBlock, fromAddr, keys, eventsPreviouslySent)
 		},
 	}
 	return h.subscribe(ctx, w, subscriber)
 }
 
+type errorTxnHashNotFound struct {
+	txHash felt.Felt
+}
```

Comment:
one suggestion
```suggestion
type errorTxnHashNotFound felt.Felt
```

One nit: move it to the top of the file where we usually have all the definitions